tests/test_checkout.rs

```diff
@@ -73,6 +73,223 @@ fn test_checkout() {
     "###);
 }
 
+#[test]
+fn test_checkout_overwrite_normal_file() {
```

Comment:
> I've generally put tests at the lowest reasonable level. For example, most testing of e.g. diffs and conflicts are done at a low level and the CLI tests only test a few cases to make sure that things are integrated and rendered correctly.

I tend to agree with this test strategy. I put these tests to CLI layer just because I'm not familiar with the internal APIs, and as you said, it's easy to write CLI tests.
Since the maintenance cost of CLI tests would be higher, it's probably better to write this kind of tests in lib.